### PR TITLE
fix: Add french translation missing

### LINKED ISSUE #20060: Missing French Translation
Issues to add french translation missing

Step : 

- [ ]  The additional French translations
- [ ] Fixing the untranslated strings
- [ ] using gettext instead of lazy_gettext
- [ ] translate variable
- [ ] translate changed_on_delta_humanized

### PR DESCRIPTION
**Pull request by aehanno**
### SUMMARY
Adding french translation 

### Add Information
Fixes https://github.com/apache/superset/issues/20060

### CODE REVIEW

File: superset-frontend/src/dashboard/components/SliceAdder.jsx [hunk 1]
```diff
@@ -253,7 +254,7 @@ class SliceAdder extends React.Component {
             value={this.state.sortBy}
             onChange={this.handleSelect}
             options={Object.entries(KEYS_TO_SORT).map(([key, label]) => ({
-              label: t('Sort by %s', label),
+              label: t('Sort by %s', t(label)),
```
Comment (villebro) [inline superset-frontend/src/dashboard/components/SliceAdder.jsx:257]: This doesn't look right; you can't translate variables like this (the original code looks correct)

File: superset-frontend/src/dashboard/components/SliceAdder.jsx [hunk 2]
```diff
@@ -221,7 +222,7 @@ class SliceAdder extends React.Component {
             innerRef={dragSourceRef}
             style={style}
             sliceName={cellData.slice_name}
-            lastModified={cellData.changed_on_humanized}
+            lastModified={moment.utc(cellData.changed_on).fromNow()}
```
Comment (villebro) [inline superset-frontend/src/dashboard/components/SliceAdder.jsx:225]: This should not be changed (@dpgaspar can comment in more detail, but IIRC, the `changed_on` property needs to be constructed in the backend right now)

File: superset-frontend/src/dashboard/components/dnd/AddSliceDragPreview.jsx [hunk 3]
```diff
@@ -72,7 +73,7 @@ function AddSliceDragPreview({ dragItem, slices, isDragging, currentOffset }) {
         transform: `translate(${currentOffset.x}px, ${currentOffset.y}px)`,
       }}
       sliceName={slice.slice_name}
-      lastModified={slice.changed_on_humanized}
+      lastModified={moment.utc(slice.changed_on).fromNow()}
```
Comment (villebro) [inline superset-frontend/src/dashboard/components/dnd/AddSliceDragPreview.jsx:76]: same here

File: superset-frontend/src/explore/components/controls/DateFilterControl/DateFilterLabel.tsx [hunk 4]
```diff
@@ -314,7 +314,7 @@ export default function DateFilterLabel(props: DateFilterControlProps) {
       <Divider />
       <div>
         <div className="section-title">{t('Actual time range')}</div>
-        {validTimeRange && <div>{evalResponse}</div>}
+        {validTimeRange && <div>{t(evalResponse)}</div>}
```
Comment (villebro) [inline superset-frontend/src/explore/components/controls/DateFilterControl/DateFilterLabel.tsx:317]: Again, you can't translate a variable (please see e.g. https://stackoverflow.com/questions/34579316/flask-babel-how-to-translate-variables for an explanation why this is the case)

File: superset-frontend/src/views/CRUD/welcome/Welcome.tsx [hunk 5]
```diff
@@ -279,7 +279,7 @@ function Welcome({ user, addDangerToast }: WelcomeProps) {
   return (
     <WelcomeContainer>
       <WelcomeNav>
-        <h1 className="welcome-header">Home</h1>
+        <h1 className="welcome-header">{t('Home')}</h1>
```
Comment (villebro) [inline superset-frontend/src/views/CRUD/welcome/Welcome.tsx:282]: Thanks for catching this!

**Review by villebro — COMMENTED** [review]
Thanks for the contributon @aehanno . I would recommend splitting this PR into multiple separate PRs, as there are many unrelated changes here. Specifically:
- The additional French translations (completely valid)
- Fixing the untranslated strings, like "Home" (this is completely valid)
- using `gettext` instead of `lazy_gettext` (this is probably ok, but should be addressed in a separate PR, perhaps fixing other simiar instances)
- Translating variables (these won't work)
- Using `moment` to calculate the `lastModified` value fron `changed_on` instead of `changed_on_humanized` + adding `last_run` (this is controversial, but we can probably have that discussion in a separate PR)

File: superset-frontend/src/views/CRUD/welcome/Welcome.tsx (hunk 5 above)
Comment (aehanno) [reply superset-frontend/src/views/CRUD/welcome/Welcome.tsx]: correction is in this PR : https://github.com/apache/superset/pull/20078

File: superset-frontend/src/explore/components/controls/DateFilterControl/DateFilterLabel.tsx (hunk 4 above)
Comment (aehanno) [reply superset-frontend/src/explore/components/controls/DateFilterControl/DateFilterLabel.tsx]: chanhe has been move in this PR : https://github.com/apache/superset/pull/20080
We can discuss about translation of varaible in the ohter PR
Without the t() here, there are any translation, do you have an idea, where I can translate that

File: superset-frontend/src/dashboard/components/SliceAdder.jsx (hunk 1 above)
Comment (aehanno) [reply superset-frontend/src/dashboard/components/SliceAdder.jsx]: PR: https://github.com/apache/superset/pull/20080

File: superset-frontend/src/dashboard/components/SliceAdder.jsx (hunk 2 above)
Comment (aehanno) [reply superset-frontend/src/dashboard/components/SliceAdder.jsx]: changed_on_humanized have any translation. I will be glade to keep 'changed_on_humanized' but what I can change to have transaltion on it

I move the change on this PR : https://github.com/apache/superset/pull/20082
we can discuss about that on the new PR

**Comment by aehanno** [comment]
Hello @villebro, 
I created 4 other pr from this one
I pinned you on each one, I hope to get your feedback

**Review by villebro — APPROVED** [review]
LGTM, thanks for adding these missing translations!

**Comment by villebro** [comment]
@mathieudruart it appears you need to fix some linting errors in the translation file:

image

**Comment by aehanno** [comment]
hey @villebro, correction of workflow has been made, it's possible to launch it ? 
Thanks

**Comment by mathieudruart** [comment]
> @mathieudruart it appears you need to fix some linting errors in the translation file:
> 
> image

Hi @villebro it should be fixed, is it possible to rerun the workflows ? Thanks !

**Comment by mathieudruart** [comment]
Hi @villebro do you know when this PR will be merged ? thank you !

**Comment by villebro** [comment]
Hey @mathieudruart , sorry for the delay; merged!

**Pull request merged**